diff --git a/proxmox-sys/src/fs/dir.rs b/proxmox-sys/src/fs/dir.rs
index c903ab87..0e8611bc 100644
--- a/proxmox-sys/src/fs/dir.rs
+++ b/proxmox-sys/src/fs/dir.rs
@@ -208,7 +208,7 @@ pub fn make_tmp_dir<P: AsRef<Path>>(
     // Push NULL byte so that we have a proper NULL-terminated string
     template.push(0);
 
-    let returned_buffer = unsafe { libc::mkdtemp(template.as_mut_ptr() as *mut i8) };
+    let returned_buffer = unsafe { libc::mkdtemp(template.as_mut_ptr() as *mut u8) };
 
     // Check errno immediately, so that nothing else can overwrite it.
     let err = std::io::Error::last_os_error();
