diff --git a/docs/Makefile b/docs/Makefile
index 9aac52e8..24282e83 100644
--- a/docs/Makefile
+++ b/docs/Makefile
@@ -67,16 +67,16 @@ ALLSPHINXOPTS   = -d $(BUILDDIR)/doctrees $(SPHINXOPTS) .
 all: ${MAN1_PAGES} ${MAN5_PAGES}
 
 config/%/config.rst: ${COMPILEDIR}/docgen
-	${COMPILEDIR}/docgen $*.cfg >$@
+	qemu-aarch64 ${COMPILEDIR}/docgen $*.cfg >$@
 
 config/acl/roles.rst: ${COMPILEDIR}/docgen
-	${COMPILEDIR}/docgen "config::acl::Role" >$@
+	qemu-aarch64 ${COMPILEDIR}/docgen "config::acl::Role" >$@
 
 %/synopsis.rst: ${COMPILEDIR}/%
-	$< printdoc > $@
+	qemu-aarch64 $< printdoc > $@
 
 proxmox-backup-client/catalog-shell-synopsis.rst: ${COMPILEDIR}/dump-catalog-shell-cli
-	${COMPILEDIR}/dump-catalog-shell-cli > proxmox-backup-client/catalog-shell-synopsis.rst
+	qemu-aarch64 ${COMPILEDIR}/dump-catalog-shell-cli > proxmox-backup-client/catalog-shell-synopsis.rst
 
 ${MAN1_PAGES} ${MAN5_PAGES}: man-pages
 
@@ -91,7 +91,7 @@ onlinehelpinfo:
 	@echo "Build finished. OnlineHelpInfo.js is in $(BUILDDIR)/scanrefs."
 
 api-viewer/apidata.js: ${COMPILEDIR}/docgen
-	${COMPILEDIR}/docgen apidata.js >$@
+	qemu-aarch64 ${COMPILEDIR}/docgen apidata.js >$@
 
 api-viewer/apidoc.js: ${API_VIEWER_FILES}
 	cat ${API_VIEWER_FILES} >$@.tmp
diff --git a/pxar-bin/tests/pxar.rs b/pxar-bin/tests/pxar.rs
index 2da5884a..29851db8 100644
--- a/pxar-bin/tests/pxar.rs
+++ b/pxar-bin/tests/pxar.rs
@@ -17,7 +17,8 @@ fn pxar_create_and_extract() {
 
     println!("run '{exec_path} create archive.pxar {src_dir}'");
 
-    Command::new(&exec_path)
+    Command::new("qemu-aarch64")
+        .arg(&exec_path)
         .arg("create")
         .arg("./tests/archive.pxar")
         .arg(src_dir)
@@ -26,7 +27,8 @@ fn pxar_create_and_extract() {
 
     println!("run '{exec_path} extract archive.pxar {dest_dir}'");
 
-    Command::new(&exec_path)
+    Command::new("qemu-aarch64")
+        .arg(&exec_path)
         .arg("extract")
         .arg("./tests/archive.pxar")
         .arg("--target")
@@ -93,7 +95,8 @@ fn pxar_split_archive_test() {
 
     println!("run '{exec_path} create archive.mpxar {src_dir} --payload-output archive.ppxar'");
 
-    Command::new(&exec_path)
+    Command::new("qemu-aarch64")
+        .arg(&exec_path)
         .arg("create")
         .arg("./tests/archive.mpxar")
         .arg(src_dir)
@@ -101,7 +104,8 @@ fn pxar_split_archive_test() {
         .status()
         .unwrap_or_else(|err| panic!("Failed to invoke '{exec_path}': {err}"));
 
-    let output = Command::new(&exec_path)
+    let output = Command::new("qemu-aarch64")
+        .arg(&exec_path)
         .arg("list")
         .arg("./tests/archive.mpxar")
         .arg("--payload-input=./tests/archive.ppxar")
@@ -122,7 +126,8 @@ fn pxar_split_archive_test() {
 
     println!("run '{exec_path} extract archive.mpxar {dest_dir} --payload-input archive.ppxar'");
 
-    Command::new(&exec_path)
+    Command::new("qemu-aarch64")
+        .arg(&exec_path)
         .arg("extract")
         .arg("./tests/archive.mpxar")
         .arg("--payload-input=./tests/archive.ppxar")
