diff --git a/Makefile b/Makefile
index 569c7e79..e3c82e26 100644
--- a/Makefile
+++ b/Makefile
@@ -10,27 +10,16 @@ SUBDIRS := etc www docs templates
 # Binaries usable by users
 USR_BIN := \
 	proxmox-backup-client 	\
-	proxmox-file-restore	\
-	pxar			\
-	proxmox-tape		\
-	pmtx			\
-	pmt
+	pxar
 
 # Binaries usable by admins
-USR_SBIN := \
-	proxmox-backup-manager \
-	proxmox-backup-debug
+USR_SBIN :=
 
 # Binaries for services:
-SERVICE_BIN := \
-	proxmox-backup-api \
-	proxmox-backup-banner \
-	proxmox-backup-proxy \
-	proxmox-daily-update
+SERVICE_BIN :=
 
 # Single file restore daemon
-RESTORE_BIN := \
-	proxmox-restore-daemon
+RESTORE_BIN :=
 
 SUBCRATES != cargo metadata --no-deps --format-version=1 \
 	| jq -r .workspace_members'[]' \
@@ -79,8 +68,7 @@ STATIC_CLIENT_DEB=$(PACKAGE)-client-static_$(DEB_VERSION)_$(ARCH).deb
 STATIC_CLIENT_DBG_DEB=$(PACKAGE)-client-static-dbgsym_$(DEB_VERSION)_$(ARCH).deb
 DOC_DEB=$(PACKAGE)-docs_$(DEB_VERSION)_all.deb
 
-DEBS=$(SERVER_DEB) $(SERVER_DBG_DEB) $(CLIENT_DEB) $(CLIENT_DBG_DEB) \
-     $(RESTORE_DEB) $(RESTORE_DBG_DEB) $(STATIC_CLIENT_DEB) $(STATIC_CLIENT_DBG_DEB)
+DEBS=${CLIENT_DEB} ${CLIENT_DBG_DEB}
 
 DSC = rust-$(PACKAGE)_$(DEB_VERSION).dsc
 
diff --git a/debian/control b/debian/control
index bd22d453..f9b5180c 100644
--- a/debian/control
+++ b/debian/control
@@ -20,7 +20,7 @@ Build-Depends: bash-completion,
                proxmox-biome:native,
                python3-docutils,
                python3-pygments,
-               python3-sphinx:all <!nodoc>,
+               python3-sphinx:all,
                rsync,
                texlive-fonts-extra <!nodoc>,
                texlive-fonts-recommended <!nodoc>,
@@ -34,35 +34,6 @@ Vcs-Browser: https://git.proxmox.com/?p=proxmox-backup.git;a=summary
 Homepage: https://www.proxmox.com
 Rules-Requires-Root: binary-targets
 
-Package: proxmox-backup-server
-Architecture: any
-Depends: fonts-font-awesome,
-         gdisk,
-         libjs-extjs (>= 7~),
-         libjs-qrcodejs (>= 1.20201119),
-         libproxmox-acme-plugins,
-         libsgutils2-1.46-2,
-         libzstd1 (>= 1.3.8),
-         lvm2,
-         openssh-server,
-         pbs-i18n,
-         postfix | mail-transport-agent,
-         proxmox-backup-docs,
-         proxmox-mini-journalreader,
-         proxmox-widget-toolkit (>= 4.3.3),
-         pve-xtermjs (>= 4.7.0-1),
-         sg3-utils,
-         smartmontools,
-         ${misc:Depends},
-         ${shlibs:Depends},
-Recommends: ifupdown2,
-            proxmox-mail-forward,
-            proxmox-offline-mirror-helper,
-            zfsutils-linux,
-Description: Proxmox Backup Server daemon with tools and GUI
- This package contains the Proxmox Backup Server daemons and related
- tools. This includes a web-based graphical user interface.
-
 Package: proxmox-backup-client
 Architecture: any
 Depends: qrencode, ${misc:Depends}, ${shlibs:Depends},
@@ -77,21 +48,3 @@ Conflicts: proxmox-backup-client,
 Description: Proxmox Backup Client tools (statically linked)
  This package contains the Proxmox Backup client, which provides a
  simple command line tool to create and restore backups.
-
-Package: proxmox-backup-docs
-Build-Profiles: <!nodoc>
-Section: doc
-Depends: fonts-font-awesome, libjs-extjs, libjs-mathjax, ${misc:Depends},
-Architecture: all
-Description: Proxmox Backup Documentation
- This package contains the Proxmox Backup Documentation files.
-
-Package: proxmox-backup-file-restore
-Architecture: any
-Depends: ${misc:Depends}, ${shlibs:Depends},
-Recommends: proxmox-backup-restore-image, pve-qemu-kvm (>= 5.0.0-9),
-Breaks: proxmox-backup-restore-image (<< 0.3.1),
-Description: Proxmox Backup single file restore tools for pxar and block device backups
- This package contains the Proxmox Backup single file restore client for
- restoring individual files and folders from both host/container and VM/block
- device backups. It includes a block device restore driver using QEMU.
diff --git a/debian/rules b/debian/rules
index 723b714f..6318e5e9 100755
--- a/debian/rules
+++ b/debian/rules
@@ -45,7 +45,7 @@ override_dh_auto_build:
 	    LIBDIR=/usr/lib/$(DEB_HOST_MULTIARCH)
 
 override_dh_missing:
-	dh_missing --fail-missing
+	#dh_missing --fail-missing
 
 override_dh_auto_install:
 	dh_auto_install -- \
@@ -55,11 +55,6 @@ override_dh_auto_install:
 	mv debian/tmp/usr/bin/proxmox-backup-client-static debian/proxmox-backup-client-static/usr/bin/proxmox-backup-client
 	mv debian/tmp/usr/bin/pxar-static debian/proxmox-backup-client-static/usr/bin/pxar
 
-override_dh_installsystemd:
-	dh_installsystemd -pproxmox-backup-server  proxmox-backup-daily-update.timer
-	# note: we start/try-reload-restart services manually in postinst
-	dh_installsystemd --no-start --no-restart-after-upgrade --no-stop-on-upgrade
-
 override_dh_fixperms:
 	dh_fixperms --exclude sg-tape-cmd
 
diff --git a/docs/Makefile b/docs/Makefile
index 2e1de4e9..bf6e8df2 100644
--- a/docs/Makefile
+++ b/docs/Makefile
@@ -1,60 +1,18 @@
 include ../defines.mk
 
 GENERATED_SYNOPSIS := 						\
-	config/acl/roles.rst					\
-	config/datastore/config.rst				\
-	config/domains/config.rst				\
-	config/media-pool/config.rst				\
-	config/notifications-priv/config.rst			\
-	config/notifications/config.rst				\
-	config/remote/config.rst				\
-	config/sync/config.rst					\
-	config/tape-job/config.rst				\
-	config/tape/config.rst					\
-	config/user/config.rst					\
-	config/verification/config.rst				\
-	config/prune/config.rst				\
-	pmt/synopsis.rst					\
-	pmtx/synopsis.rst					\
 	proxmox-backup-client/catalog-shell-synopsis.rst 	\
 	proxmox-backup-client/synopsis.rst			\
-	proxmox-backup-debug/synopsis.rst			\
-	proxmox-backup-manager/synopsis.rst			\
-	proxmox-file-restore/synopsis.rst			\
-	proxmox-tape/synopsis.rst				\
 	pxar/synopsis.rst					\
 
 MAN1_PAGES := 				\
-	pbs2to3.1			\
-	pbs3to4.1			\
-	pmt.1				\
-	pmtx.1				\
-	proxmox-backup-client.1		\
-	proxmox-backup-debug.1		\
-	proxmox-backup-manager.1	\
-	proxmox-backup-proxy.1		\
-	proxmox-file-restore.1		\
-	proxmox-tape.1			\
+ 	proxmox-backup-client.1		\
 	pxar.1				\
 
 # FIXME: prefix all man pages that are not directly relating to an existing executable with
 # `proxmox-backup.`, like the newer added proxmox-backup.node.cfg but add backwards compatible
 # symlinks, e.g. with a "5pbs" man page "suffix section".
 MAN5_PAGES :=				\
-	acl.cfg.5			\
-	datastore.cfg.5			\
-	domains.cfg.5			\
-	media-pool.cfg.5		\
-	proxmox-backup.node.cfg.5	\
-	notifications-priv.cfg.5	\
-	notifications.cfg.5		\
-	remote.cfg.5			\
-	sync.cfg.5			\
-	tape-job.cfg.5			\
-	tape.cfg.5			\
-	user.cfg.5			\
-	verification.cfg.5		\
-	prune.cfg.5				\
 
 PRUNE_SIMULATOR_FILES := 					\
 	prune-simulator/index.html				\
