diff --git a/Makefile b/Makefile
index b307009d..45b2ac3c 100644
--- a/Makefile
+++ b/Makefile
@@ -37,7 +37,7 @@ SUBCRATES != cargo metadata --no-deps --format-version=1 \
 
 ifeq ($(BUILD_MODE), release)
 CARGO_BUILD_ARGS += --release
-COMPILEDIR := target/release
+COMPILEDIR := target/aarch64-unknown-linux-gnu/release
 else
 COMPILEDIR := target/debug
 endif
diff --git a/docs/Makefile b/docs/Makefile
index b06badff..a5eb4ce2 100644
--- a/docs/Makefile
+++ b/docs/Makefile
@@ -85,7 +85,7 @@ SPHINXBUILD   = sphinx-build
 BUILDDIR      = output
 
 ifeq ($(BUILD_MODE), release)
-COMPILEDIR := ../target/release
+COMPILEDIR := ../target/aarch64-unknown-linux-gnu/release
 SPHINXOPTS    += -t release
 else
 COMPILEDIR := ../target/debug
diff --git a/pxar-bin/tests/pxar.rs b/pxar-bin/tests/pxar.rs
index f2748598..7cbdcc7a 100644
--- a/pxar-bin/tests/pxar.rs
+++ b/pxar-bin/tests/pxar.rs
@@ -8,9 +8,9 @@ fn pxar_create_and_extract() {
     let dest_dir = "../tests/catar_data/test_xattrs_dest/";
 
     let exec_path = if cfg!(debug_assertions) {
-        "../target/debug/pxar"
+        "../target/aarch64-unknown-linux-gnu/debug/pxar"
     } else {
-        "../target/release/pxar"
+        "../target/aarch64-unknown-linux-gnu/release/pxar"
     };
 
     println!("run '{} create archive.pxar {}'", exec_path, src_dir);
