diff --git a/Makefile b/Makefile
index 18e5ce27..65cf9e8d 100644
--- a/Makefile
+++ b/Makefile
@@ -2,8 +2,8 @@ include /usr/share/dpkg/pkg-info.mk
 
 PACKAGE=proxmox-biome
 
-DEB=$(PACKAGE)_$(DEB_VERSION_UPSTREAM_REVISION)_amd64.deb
-DEB_DBG=$(PACKAGE)-dbgsym_$(DEB_VERSION_UPSTREAM_REVISION)_amd64.deb
+DEB=$(PACKAGE)_$(DEB_VERSION_UPSTREAM_REVISION)_arm64.deb
+DEB_DBG=$(PACKAGE)-dbgsym_$(DEB_VERSION_UPSTREAM_REVISION)_arm64.deb
 
 BUILDDIR=$(PACKAGE)-$(DEB_VERSION_UPSTREAM)
 
@@ -45,7 +45,7 @@ $(BUILDDIR):
 
 deb: $(DEB)
 $(DEB) $(DEB_DBG) &: $(BUILDDIR)
-	cd $(BUILDDIR); dpkg-buildpackage -b -us -uc
+	cd $(BUILDDIR); dpkg-buildpackage -aarm64 -b -us -uc
 	lintian $(DEB)
 
 .PHONY: upload
