diff --git a/.cargo/config b/.cargo/config
index 3b5b6e4..e6d9429 100644
--- a/.cargo/config
+++ b/.cargo/config
@@ -1,5 +1,5 @@
 [source]
-[source.debian-packages]
-directory = "/usr/share/cargo/registry"
+[source.vendored-sources]
+directory = "vendor"
 [source.crates-io]
-replace-with = "debian-packages"
+replace-with = "vendored-sources"
diff --git a/Cargo.toml b/Cargo.toml
index b49a591..13147d6 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -6,7 +6,14 @@ members = [
 ]
 
 [patch.crates-io]
-# proxmox-tfa requires -time and -uuid as well, so enable *all* or *none* of them
-#proxmox-tfa = { path = "../proxmox/proxmox-tfa" }
-#proxmox-time = { path = "../proxmox/proxmox-time" }
-#proxmox-uuid = { path = "../proxmox/proxmox-uuid" }
+perlmod = { path = "perlmod/perlmod" }
+proxmox-acme-rs = { path = "proxmox-acme-rs" }
+proxmox-apt = { path = "proxmox-apt" }
+proxmox-http = { path = "proxmox/proxmox-http" }
+proxmox-openid = { path = "proxmox-openid-rs" }
+proxmox-resource-scheduling = { path = "proxmox-resource-scheduling" }
+proxmox-schema = { path = "proxmox/proxmox-schema" }
+proxmox-subscription = { path = "proxmox/proxmox-subscription" }
+proxmox-sys = { path = "proxmox/proxmox-sys" }
+proxmox-tfa = { path = "proxmox/proxmox-tfa" }
+proxmox-time = { path = "proxmox/proxmox-time" }
diff --git a/Makefile b/Makefile
index 3ddafd0..e6d5c04 100644
--- a/Makefile
+++ b/Makefile
@@ -68,12 +68,18 @@ gen:
 build:
 	rm -rf build
 	mkdir build
-	echo system >build/rust-toolchain
+	#echo system >build/rust-toolchain
 	cp -a ./scripts ./build
 	cp -a ./common ./build
 	cp -a ./pve-rs ./build
 	cp -a ./pmg-rs ./build
 	cp -a ./Proxmox ./build
+	ln -s ../perlmod ./build/perlmod
+	ln -s ../proxmox ./build/proxmox
+	ln -s ../proxmox-apt ./build/proxmox-apt
+	ln -s ../proxmox-acme-rs ./build/proxmox-acme-rs
+	ln -s ../proxmox-openid-rs ./build/proxmox-openid-rs
+	ln -s ../proxmox-resource-scheduling ./build/proxmox-resource-scheduling
 	$(MAKE) BUILD_MODE=release -C build -f ../Makefile gen
 	mkdir -p ./build/pve-rs/Proxmox/Lib
 	mv ./build/Proxmox/Lib/PVE.pm ./build/pve-rs/Proxmox/Lib/PVE.pm
diff --git a/pmg-rs/Cargo.toml b/pmg-rs/Cargo.toml
index 903ed3b..0ea6a12 100644
--- a/pmg-rs/Cargo.toml
+++ b/pmg-rs/Cargo.toml
@@ -39,3 +39,14 @@ proxmox-subscription = "0.3"
 proxmox-sys = "0.4"
 proxmox-tfa = { version = "2.1", features = ["api"] }
 proxmox-time = "1.1.3"
+
+[patch.crates-io]
+perlmod = { path = "../perlmod/perlmod" }
+proxmox-acme-rs = { path = "../proxmox-acme-rs" }
+proxmox-apt = { path = "../proxmox-apt" }
+proxmox-http = { path = "../proxmox/proxmox-http" }
+proxmox-schema = { path = "../proxmox/proxmox-schema" }
+proxmox-subscription = { path = "../proxmox/proxmox-subscription" }
+proxmox-sys = { path = "../proxmox/proxmox-sys" }
+proxmox-tfa = { path = "../proxmox/proxmox-tfa" }
+proxmox-time = { path = "../proxmox/proxmox-time" }
diff --git a/pmg-rs/debian/control b/pmg-rs/debian/control
index 5af088b..00df9d6 100644
--- a/pmg-rs/debian/control
+++ b/pmg-rs/debian/control
@@ -4,33 +4,7 @@ Priority: optional
 Maintainer: Proxmox Support Team <support@proxmox.com>
 Build-Depends:
  debhelper (>= 12),
- dh-cargo (>= 24),
- cargo:native <!nocheck>,
- rustc:native <!nocheck>,
  libstd-rust-dev <!nocheck>,
- librust-anyhow-1+default-dev,
- librust-hex-0.4+default-dev,
- librust-http-0.2+default-dev (>= 0.2.7-~~),
- librust-libc-0.2+default-dev,
- librust-nix-0.24+default-dev,
- librust-openssl-0.10+default-dev (>= 0.10.40-~~),
- librust-perlmod-0.13+default-dev,
- librust-perlmod-0.13+exporter-dev,
- librust-proxmox-acme-rs-0.4+client-dev,
- librust-proxmox-acme-rs-0.4+default-dev,
- librust-proxmox-apt-0.9+default-dev,
- librust-proxmox-http-0.7+client-sync-dev,
- librust-proxmox-http-0.7+client-trait-dev,
- librust-proxmox-http-0.7+default-dev,
- librust-proxmox-subscription-0.3+default-dev,
- librust-proxmox-sys-0.4+default-dev,
- librust-proxmox-tfa-2+api-dev (>= 2.1-~~),
- librust-proxmox-tfa-2+default-dev (>= 2.1-~~),
- librust-proxmox-time-1+default-dev (>= 1.1.3-~~),
- librust-serde-1+default-dev,
- librust-serde-bytes-0.11+default-dev (>= 0.11.3-~~),
- librust-serde-json-1+default-dev,
- librust-url-2+default-dev,
 Standards-Version: 4.3.0
 Homepage: https://www.proxmox.com
 
diff --git a/pve-rs/Cargo.toml b/pve-rs/Cargo.toml
index 6a56d3a..81aeddf 100644
--- a/pve-rs/Cargo.toml
+++ b/pve-rs/Cargo.toml
@@ -38,3 +38,15 @@ proxmox-subscription = "0.3"
 proxmox-sys = "0.4"
 proxmox-tfa = { version = "2.1", features = ["api"] }
 proxmox-time = "1.1.3"
+
+[patch.crates-io]
+perlmod = { path = "../perlmod/perlmod" }
+proxmox-apt = { path = "../proxmox-apt" }
+proxmox-http = { path = "../proxmox/proxmox-http" }
+proxmox-openid = { path = "../proxmox-openid-rs" }
+proxmox-resource-scheduling = { path = "../proxmox-resource-scheduling" }
+proxmox-schema = { path = "../proxmox/proxmox-schema" }
+proxmox-subscription = { path = "../proxmox/proxmox-subscription" }
+proxmox-sys = { path = "../proxmox/proxmox-sys" }
+proxmox-tfa = { path = "../proxmox/proxmox-tfa" }
+proxmox-time = { path = "../proxmox/proxmox-time" }
diff --git a/pve-rs/debian/control b/pve-rs/debian/control
index 6120116..fa38b10 100644
--- a/pve-rs/debian/control
+++ b/pve-rs/debian/control
@@ -3,35 +3,7 @@ Section: perl
 Priority: optional
 Build-Depends:
  debhelper (>= 12),
- dh-cargo (>= 24),
- cargo:native <!nocheck>,
- rustc:native <!nocheck>,
  libstd-rust-dev <!nocheck>,
- librust-anyhow-1+default-dev,
- librust-base32-0.4+default-dev,
- librust-base64-0.13+default-dev,
- librust-hex-0.4+default-dev,
- librust-http-0.2+default-dev (>= 0.2.7-~~),
- librust-libc-0.2+default-dev,
- librust-nix-0.24+default-dev,
- librust-openssl-0.10+default-dev (>= 0.10.40-~~),
- librust-perlmod-0.13+default-dev,
- librust-perlmod-0.13+exporter-dev,
- librust-proxmox-apt-0.9+default-dev,
- librust-proxmox-http-0.7+client-sync-dev,
- librust-proxmox-http-0.7+client-trait-dev,
- librust-proxmox-http-0.7+default-dev,
- librust-proxmox-openid-0.9+default-dev (>= 0.9.5-~~),
- librust-proxmox-resource-scheduling-0.1+default-dev,
- librust-proxmox-subscription-0.3+default-dev,
- librust-proxmox-sys-0.4+default-dev,
- librust-proxmox-tfa-2+api-dev (>= 2.1-~~),
- librust-proxmox-tfa-2+default-dev (>= 2.1-~~),
- librust-proxmox-time-1+default-dev (>= 1.1.3-~~),
- librust-serde-1+default-dev,
- librust-serde-bytes-0.11+default-dev,
- librust-serde-json-1+default-dev,
- librust-url-2+default-dev,
 Maintainer: Proxmox Support Team <support@proxmox.com>
 Standards-Version: 4.5.1
 Vcs-Git: git://git.proxmox.com/git/proxmox-perl-rs.git
