diff --git a/Makefile b/Makefile
index 18e5ce27..65cf9e8d 100644
--- a/Makefile
+++ b/Makefile
@@ -2,8 +2,8 @@ include /usr/share/dpkg/pkg-info.mk
 
 PACKAGE=proxmox-biome
 
-DEB=$(PACKAGE)_$(DEB_VERSION_UPSTREAM_REVISION)_amd64.deb
-DEB_DBG=$(PACKAGE)-dbgsym_$(DEB_VERSION_UPSTREAM_REVISION)_amd64.deb
+DEB=$(PACKAGE)_$(DEB_VERSION_UPSTREAM_REVISION)_arm64.deb
+DEB_DBG=$(PACKAGE)-dbgsym_$(DEB_VERSION_UPSTREAM_REVISION)_arm64.deb
 
 BUILDDIR=$(PACKAGE)-$(DEB_VERSION_UPSTREAM)
 
@@ -45,7 +45,7 @@ $(BUILDDIR):
 
 deb: $(DEB)
 $(DEB) $(DEB_DBG) &: $(BUILDDIR)
-	cd $(BUILDDIR); dpkg-buildpackage -b -us -uc
+	cd $(BUILDDIR); dpkg-buildpackage -aarm64 -b -us -uc
 	lintian $(DEB)
 
 .PHONY: upload
diff --git a/biome/rust-toolchain.toml b/biome/rust-toolchain.toml
index fbae3fa6..7c761706 100644
--- a/biome/rust-toolchain.toml
+++ b/biome/rust-toolchain.toml
@@ -2,4 +2,4 @@
 # The default profile includes rustc, rust-std, cargo, rust-docs, rustfmt and clippy.
 # https://rust-lang.github.io/rustup/concepts/profiles.html
 profile = "default"
-channel = "1.87.0"
+channel = "stable"
diff --git a/debian/control b/debian/control
index 7109d042..80bcb58b 100644
--- a/debian/control
+++ b/debian/control
@@ -8,7 +8,7 @@ Homepage: https://biomejs.dev/
 Standards-Version: 4.7.0.0
 
 Package: proxmox-biome
-Architecture: amd64
+Architecture: arm64
 Depends: ${misc:Depends}, ${shlibs:Depends},
 Description: Biome offers JS formatter and linter, usable via CLI and LSP.
  Biome is a performant toolchain for web projects, it aims to provide developer
diff --git a/debian/install b/debian/install
index 80c48310..312138c2 100644
--- a/debian/install
+++ b/debian/install
@@ -1,3 +1,3 @@
 src/biome.json usr/share/proxmox-biome/
 src/proxmox-biome usr/bin/
-target/release/biome usr/bin
+target/aarch64-unknown-linux-gnu/release/biome usr/bin
