diff --git a/.cargo/config.toml b/.cargo/config.toml
index a439c97b..a95ae483 100644
--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -1,8 +1,20 @@
-[source]
-[source.debian-packages]
-directory = "/usr/share/cargo/registry"
 [source.crates-io]
-replace-with = "debian-packages"
+replace-with = "vendored-sources"
+
+[source."git://git.proxmox.com/git/pathpatterns.git"]
+git = "git://git.proxmox.com/git/pathpatterns.git"
+replace-with = "vendored-sources"
+
+[source."git://git.proxmox.com/git/proxmox-fuse.git"]
+git = "git://git.proxmox.com/git/proxmox-fuse.git"
+replace-with = "vendored-sources"
+
+[source."git://git.proxmox.com/git/pxar.git"]
+git = "git://git.proxmox.com/git/pxar.git"
+replace-with = "vendored-sources"
+
+[source.vendored-sources]
+directory = "vendor"
 
 [profile.release]
 debug=true
diff --git a/Makefile b/Makefile
index d7470146..e5fc29ca 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,4 @@
 include /usr/share/dpkg/default.mk
-include /usr/share/rustc/architecture.mk
 include defines.mk
 
 PACKAGE := proxmox-backup
diff --git a/debian/rules b/debian/rules
index be7c93c2..df9e8e27 100755
--- a/debian/rules
+++ b/debian/rules
@@ -4,12 +4,9 @@
 DH_VERBOSE = 1
 
 include /usr/share/dpkg/pkg-info.mk
-include /usr/share/rustc/architecture.mk
 
 export BUILD_MODE=release
 
-export CARGO=/usr/share/cargo/bin/cargo
-
 export CFLAGS CXXFLAGS CPPFLAGS LDFLAGS
 export DEB_HOST_RUST_TYPE DEB_HOST_GNU_TYPE
 export CARGO_HOME = $(CURDIR)/debian/cargo_home
@@ -27,12 +24,6 @@ endif
 override_dh_auto_configure:
 	@perl -ne 'if (/^version\s*=\s*"(\d+(?:\.\d+)+)"/) { my $$v_cargo = $$1; my $$v_deb = "$(DEB_VERSION_UPSTREAM)"; \
 	    die "ERROR: d/changelog <-> Cargo.toml version mismatch: $$v_cargo != $$v_deb\n" if $$v_cargo ne $$v_deb; exit(0); }' Cargo.toml
-	$(CARGO) prepare-debian $(CURDIR)/debian/cargo_registry --link-from-system
-	# `cargo build` and `cargo install` have different config precedence, symlink
-	# the wrapper config into a place where `build` picks it up as well..
-	# https://doc.rust-lang.org/cargo/commands/cargo-install.html#configuration-discovery
-	mkdir -p .cargo
-	ln -s $(CARGO_HOME)/config.toml $(CURDIR)/.cargo/config.toml
 	dh_auto_configure
 
 override_dh_auto_build:
