diff --git a/src/main.rs b/src/main.rs
index 7896a9c..f5dd39f 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -129,6 +129,7 @@ fn read_ticket_line(
             io_bail!("timed out");
         }
     }
+    poll.registry().deregister(stream)?;
 
     let newline_idx = &buf[..].iter().position(|&x| x == b'\n').unwrap();
 
@@ -217,6 +218,7 @@ fn listen_and_accept(
         if !events.is_empty() {
             let (stream, client) = listener.accept()?;
             println!("client connection: {:?}", client);
+            poll.registry().deregister(&mut listener)?;
             return Ok((stream, port));
         }
 
