diff --git a/.cargo/config b/.cargo/config
index 3b5b6e4..b08ce29 100644
--- a/.cargo/config
+++ b/.cargo/config
@@ -1,5 +1,5 @@
 [source]
-[source.debian-packages]
-directory = "/usr/share/cargo/registry"
+[source.vendor]
+directory = "vendor"
 [source.crates-io]
-replace-with = "debian-packages"
+replace-with = "vendor"
diff --git a/Cargo.toml b/Cargo.toml
index b9479be..0d417f4 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -19,3 +19,9 @@ clap = "3"
 proxmox-io = "1"
 proxmox-lang = "1.1"
 proxmox-sys = "0.5"
+
+
+[patch.crates-io]
+proxmox-io = { path = "../proxmox/proxmox-io" }
+proxmox-lang = { path = "../proxmox/proxmox-lang" }
+proxmox-sys = { path = "../proxmox/proxmox-sys" }
diff --git a/Makefile b/Makefile
index 01488d1..cb0ceb7 100644
--- a/Makefile
+++ b/Makefile
@@ -46,6 +46,14 @@ $(TERMPROXY_BIN): .do-cargo-build
 .PHONY: cargo-build
 cargo-build: .do-cargo-build
 
+.PHONY: cargo-vendor
+cargo-vendor:
+	cargo vendor
+	rm -f vendor/endian_trait/rust-toolchain
+	rm -f vendor/endian_trait_derive/rust-toolchain
+	sed -i "s#\"rust-toolchain[^,]\+,##"  vendor/endian_trait/.cargo-checksum.json
+	sed -i "s#\"rust-toolchain[^,]\+,##"  vendor/endian_trait_derive/.cargo-checksum.json
+
 update-dcontrol:
 	rm -rf $(BUILDDIR)
 	$(MAKE) $(BUILDDIR)
@@ -74,9 +82,10 @@ $(ORIG_SRC_TAR): $(BUILDDIR)
 .PHONY: deb
 deb: $(DEB)
 $(DBG_DEB): $(DEB)
-$(DEB): $(BUILDDIR)
-	cd $(BUILDDIR); dpkg-buildpackage -b -uc -us
-	lintian $(DEB)
+$(DEB): cargo-vendor
+	echo "git clone git://git.proxmox.com/git/pve-xtermjs.git\\ngit checkout ${GITVERSION}" > debian/SOURCE
+	DEB_CARGO_PACKAGE=$(PACKAGE) dpkg-buildpackage -b -uc -us --no-pre-clean
+	lintian ../$(DEB)
 	@echo $(DEB)
 
 .PHONY: dsc
diff --git a/debian/control b/debian/control
index fd052c2..65433dd 100644
--- a/debian/control
+++ b/debian/control
@@ -1,20 +1,8 @@
 Source: rust-termproxy
 Section: admin
 Priority: optional
-Build-Depends: cargo:native,
-               debhelper-compat (= 13),
-               dh-cargo (>= 25),
-               librust-anyhow-1+default-dev,
-               librust-clap-3+default-dev,
-               librust-mio-0.8+default-dev,
-               librust-mio-0.8+net-dev,
-               librust-mio-0.8+os-ext-dev,
-               librust-proxmox-io-1+default-dev,
-               librust-proxmox-lang-1+default-dev (>= 1.1-~~),
-               librust-proxmox-sys-0.5+default-dev,
-               librust-ureq-2+gzip-dev (>= 2.4-~~),
+Build-Depends: debhelper-compat (= 13),
                libstd-rust-dev,
-               rustc:native,
 Maintainer: Proxmox Support Team <support@proxmox.com>
 Standards-Version: 4.6.1
 Vcs-Git: git://git.proxmox.com/git/pve-xtermjs.git
diff --git a/debian/install b/debian/install
index 04be689..0e69d82 100644
--- a/debian/install
+++ b/debian/install
@@ -1 +1,2 @@
-src/www/*		/usr/share/pve-xtermjs/
+src/www/*			/usr/share/pve-xtermjs/
+target/release/termproxy	/usr/bin/
diff --git a/debian/rules b/debian/rules
index 6049291..c9fdaf4 100755
--- a/debian/rules
+++ b/debian/rules
@@ -1,7 +1,6 @@
 #!/usr/bin/make -f
 
 include /usr/share/dpkg/pkg-info.mk
-include /usr/share/rustc/architecture.mk
 
 export BUILD_MODE=release
 
@@ -23,5 +23,4 @@ override_dh_auto_build:
 
 override_dh_auto_configure:
 	rm -f Cargo.lock
-	/usr/share/cargo/bin/cargo prepare-debian $(CURDIR)/debian/cargo_registry --link-from-system
 	dh_auto_configure
diff --git a/debian/rules.debcargo.hint b/debian/rules.debcargo.hint
index 92c239b..4aaf62a 100755
--- a/debian/rules.debcargo.hint
+++ b/debian/rules.debcargo.hint
@@ -1,6 +1,6 @@
 #!/usr/bin/make -f
 %:
-	dh $@ --buildsystem cargo
+	dh $@
 
 override_dh_auto_test:
 	dh_auto_test -- test --all
