diff --git a/src/acme/client.rs b/src/acme/client.rs
index 96cee0fe..fb452b35 100644
--- a/src/acme/client.rs
+++ b/src/acme/client.rs
@@ -11,7 +11,7 @@ use nix::sys::stat::Mode;
 use serde::{Deserialize, Serialize};
 
 use proxmox_acme::account::AccountCreator;
-use proxmox_acme::account::AccountData as AcmeAccountData;
+use proxmox_acme::types::AccountData as AcmeAccountData;
 use proxmox_acme::order::{Order, OrderData};
 use proxmox_acme::Request as AcmeRequest;
 use proxmox_acme::{Account, Authorization, Challenge, Directory, Error, ErrorResponse};
diff --git a/src/api2/config/acme.rs b/src/api2/config/acme.rs
index 4fa276f6..f18a517f 100644
--- a/src/api2/config/acme.rs
+++ b/src/api2/config/acme.rs
@@ -16,7 +16,8 @@ use proxmox_router::{
 use proxmox_schema::{api, param_bail};
 use proxmox_sys::{task_log, task_warn};
 
-use proxmox_acme::account::AccountData as AcmeAccountData;
+use proxmox_acme::types::AccountData as AcmeAccountData;
+
 use proxmox_acme::Account;
 
 use pbs_api_types::{Authid, PRIV_SYS_MODIFY};
