diff --git a/Makefile b/Makefile
index b307009d..45b2ac3c 100644
--- a/Makefile
+++ b/Makefile
@@ -37,7 +37,7 @@ SUBCRATES != cargo metadata --no-deps --format-version=1 \
 
 ifeq ($(BUILD_MODE), release)
 CARGO_BUILD_ARGS += --release
-COMPILEDIR := target/release
+COMPILEDIR := target/aarch64-unknown-linux-gnu/release
 else
 COMPILEDIR := target/debug
 endif
diff --git a/docs/Makefile b/docs/Makefile
index b06badff..a5eb4ce2 100644
--- a/docs/Makefile
+++ b/docs/Makefile
@@ -85,7 +85,7 @@ SPHINXBUILD   = sphinx-build
 BUILDDIR      = output
 
 ifeq ($(BUILD_MODE), release)
-COMPILEDIR := ../target/release
+COMPILEDIR := ../target/aarch64-unknown-linux-gnu/release
 SPHINXOPTS    += -t release
 else
 COMPILEDIR := ../target/debug
@@ -98,16 +98,16 @@ ALLSPHINXOPTS   = -d $(BUILDDIR)/doctrees $(SPHINXOPTS) .
 all: ${MAN1_PAGES} ${MAN5_PAGES}
 
 config/%/config.rst: ${COMPILEDIR}/docgen
-	${COMPILEDIR}/docgen $*.cfg >$@
+	qemu-aarch64 ${COMPILEDIR}/docgen $*.cfg >$@
 
 config/acl/roles.rst: ${COMPILEDIR}/docgen
-	${COMPILEDIR}/docgen "config::acl::Role" >$@
+	qemu-aarch64 ${COMPILEDIR}/docgen "config::acl::Role" >$@
 
 %/synopsis.rst: ${COMPILEDIR}/%
-	$< printdoc > $@
+	qemu-aarch64 $< printdoc > $@
 
 proxmox-backup-client/catalog-shell-synopsis.rst: ${COMPILEDIR}/dump-catalog-shell-cli
-	${COMPILEDIR}/dump-catalog-shell-cli > proxmox-backup-client/catalog-shell-synopsis.rst
+	qemu-aarch64 ${COMPILEDIR}/dump-catalog-shell-cli > proxmox-backup-client/catalog-shell-synopsis.rst
 
 ${MAN1_PAGES} ${MAN5_PAGES}: man-pages
 
@@ -122,7 +122,7 @@ onlinehelpinfo:
 	@echo "Build finished. OnlineHelpInfo.js is in $(BUILDDIR)/scanrefs."
 
 api-viewer/apidata.js: ${COMPILEDIR}/docgen
-	${COMPILEDIR}/docgen apidata.js >$@
+	qemu-aarch64 ${COMPILEDIR}/docgen apidata.js >$@
 
 api-viewer/apidoc.js: ${API_VIEWER_FILES}
 	cat ${API_VIEWER_FILES} >$@.tmp
diff --git a/pxar-bin/tests/pxar.rs b/pxar-bin/tests/pxar.rs
index f2748598..0c31ee4f 100644
--- a/pxar-bin/tests/pxar.rs
+++ b/pxar-bin/tests/pxar.rs
@@ -8,14 +8,15 @@ fn pxar_create_and_extract() {
     let dest_dir = "../tests/catar_data/test_xattrs_dest/";
 
     let exec_path = if cfg!(debug_assertions) {
-        "../target/debug/pxar"
+        "../target/aarch64-unknown-linux-gnu/debug/pxar"
     } else {
-        "../target/release/pxar"
+        "../target/aarch64-unknown-linux-gnu/release/pxar"
     };
 
     println!("run '{} create archive.pxar {}'", exec_path, src_dir);
 
-    Command::new(exec_path)
+    Command::new("qemu-aarch64")
+        .arg(exec_path)
         .arg("create")
         .arg("./tests/archive.pxar")
         .arg(src_dir)
@@ -24,7 +25,8 @@ fn pxar_create_and_extract() {
 
     println!("run '{} extract archive.pxar {}'", exec_path, dest_dir);
 
-    Command::new(exec_path)
+    Command::new("qemu-aarch64")
+        .arg(exec_path)
         .arg("extract")
         .arg("./tests/archive.pxar")
         .arg("--target")
